cmse/cmse.md

```diff
@@ -0,0 +1,1725 @@
+in Armv8-M. This instruction is not part of CMSE, but is closely related.
+
+Some of the requirements defined by this document will be included in future
+[[ACLE]](#ACLE) and [[BSABI]](#BSABI) documents.
+
+## Conventions
+
+This document consists of informative text and requirements.
+
+> 0	Requirements are numbered in the left margin and highlighted as shown here.
+
+A permanent unique reference consists of the document number, document version,
+and requirement number.
+
+# OVERVIEW OF CMSE
+
+## Introduction
+
+CMSE is an optional part of the Armv8-M baseline and mainline architecture 
+defined by [[ARMV8M]](#ARMV8M). CMSE is designed to combine code from multiple
+vendors without requiring trust between them. CMSE gives vendors the ability to
+protect their software assets (code and data) by restricting access to the
+memory where their software assets reside, except for a set of explicitly
+exported entry points that are defined by the vendor. This supports the creation
+of a trusted software stack that provides features such as secure remote
+firmware updates, while significantly reducing the attack surface of such code.
+This is an important feature for any network-connected device that can be
+updated after deployment, including any IoT device.
+
+CMSE defines a system-wide division of physical memory into _secure regions_ and
+_non-secure regions_ and two system-wide _security states_ that are enforced by
+hardware. There is a direct relation between the memory regions and the security
+states:
+
+*	Code executed from a non-secure region (_non-secure code_) is executed in
+  _non-secure state_ and can only access memory in non-secure regions.
+*	Code executed from a secure region (_secure code_) is executed in 
+  _secure state_ and can access memory in both secure and non-secure
+  regions.[ ]
+
+Attempts to access secure regions from non-secure code or a mismatch between
+the (secure or non-secure) code that is executed and the security state of the
+system results in a _SecureFault_.
+
+The security states are orthogonal to the exception level, as shown in 
+[Figure 1](#figure1).
+
+![](figures/diag-rep.png)
+
+<span id="figure1" class="citation-label">**Figure 1: Diagrammatic representation of secure states**</span>
+
+Memory regions can be defined by the system through the _IDAU_ or can be
+controlled in software through the memory mapped _SAU_ registers.
+
+Parts of the system are banked between the security states. The stack pointer is
+banked, resulting in a stack pointer for each combination of security state and
+exception level. All parts of the system accessible in non-secure state can be
+accessed in secure state as well, including the banked parts.
+
+###  Security state changes
+
+The system boots in secure state and can change security states using branches
+as summarized in [Figure 2](#figure2).
+
+![](figures/state-trans.png)
+
+<span id="figure2" class="citation-label">**Figure 2: Security state transitions**</span>
```

Comment:
Can this be done this way? So that the caption goes with the image itself, and the links to `figure2` will point at the caption.

```suggestion
![<span id="figure2" class="citation-label">**Figure 2: Security state transitions**</span>
](figures/state-trans.png)
```

Reply:
As mentioned below, the figure gets placed on a different page if done that way.

Reply:
That's not an issue. The figure is linked via the `<span...`. Please use the `<span...` as the caption of the figures.

Reply:
Oh, also, it doesn't appear as text below the figure on the website. https://valeriadaneva.github.io/acle/cmse/cmse.html#figure1